docs/arcadia-users-group/20240416-intro-to-ml1/make_peptides_ml_dataset.md

```diff
@@ -0,0 +1,278 @@
+   current_table = None
+   output_file = None
+   
+   with open('backup_dump.sql', 'r') as dump_file:
+       for line in dump_file:
+           # Check for COPY command start
+           if line.startswith('COPY'):
+               table_name = line.split()[1]  # Assumes table name is the second word
+               current_table = table_name
+               output_file = open(f'{output_file_prefix}{current_table}.csv', 'w')
+           elif line.startswith('\.'):
+               # End of data for the current table
+               if output_file:
+                   output_file.close()
+                   output_file = None
+           elif output_file:
+               # Convert tab-separated values to comma-separated values
+               csv_line = line.replace('\t', ',')
+               output_file.write(csv_line)
+   ```
+
+4. Filter to peptides that are at least k in length. MMSeqs2 clustering in step 4 will fail if any of the sequences are below the k-mer length used for clustering.
+   ```bash
+   seqkit seq --min-len 10 -o peptipedia_minlen10.fasta peptipedia.fasta
+   ```
+
+5. Cluster the sequences.
+   Using a low identity threshold (10%), this step determines which peptides in the input database have similar sequences.
+   We will subsequently use this information to artificially generate a data set where the peptides or their characteristics are predictive of bioactivity labels.
+   ```bash
+   mmseqs easy-cluster peptipedia_minlen10.fasta mmseqs/peptipedia_minlen10-0.1 tmp --min-seq-id 0.1
+   ```
+
+6. Filter to large clusters composed of peptides that all have the same bioactivity prediction and write a TSV file.
+   Many of the peptides that end up in clusters have multiple bioactivity predictions (e.g. antimicrobial and therapeutic).
+   The script below filters to peptide clusters that all have at least one bioactivity label that is the same.
+   It also adds additional metadata (GO, PFAM) from the SQL database to the data.frame of peptide information.
+   ```r
+   library(tidyverse)
+   library(janitor)
+   
+   # The goal of this script is to create a demo machine learning data set from peptides and their metadata.
+   # It filters peptides in the peptipedia database to large clusters of similar peptides
+   # (80% identity or greater) that all have the same bioactivity label.
+   # It then joins selected peptides to their metadata.
+   
+   # filter to peptides clusters with many members that have the same bioactivity --------
+
+   # read in cluster information from mmseqs2 linclust
+   clusters <- read_tsv("~/Downloads/backup_peptipedia_29_03_2023/mmseqs/peptipedia_minlen10-0.8_cluster.tsv",
+                        col_names = c("rep", "member"))
+   
+   # read in peptide bioactivity information
+   peptide_bioactivity <- read_csv('~/Downloads/backup_peptipedia_29_03_2023/peptipedia.csv') %>%
+     clean_names()
+   
+   # make the data set a little bit smaller by filtering to clusters that have > 5 members
```

Comment:
code says `> 2`

Reply:
thank youuuu. I'll fix this. I was experimenting with different thresholds and 2 ended up working the best. Good catch!